AK/Format.h

```diff
@@ -624,6 +624,16 @@ struct Formatter<Error> : Formatter<FormatString> {
     }
 };
 
+template<typename T, typename ErrorType>
+struct Formatter<ErrorOr<T, ErrorType>> : Formatter<FormatString> {
+    ErrorOr<void> format(FormatBuilder& builder, ErrorOr<T, ErrorType> const& error_or)
+    {
+        if (error_or.is_error())
+            return Formatter<FormatString>::format(builder, "{}", error_or.error());
```

Comment:
This creates a temporary `StringBuilder`, which might allocate memory.
 Would something like the following work?
```diff
            return Formatter<ErrorType>::format(builder, error_or.error());
```

Reply:
Oh yeah, you're right!

Reply:
`Formatter<FormatString>::format`calls the format method of the superclass, on the same instance as `this`. However, `Formatter<ErrorOr<…>>` is not a superclass, and we don't really want multiple inheritance here.

Since we already do lots of allocations anyway, and printing ErrorOr's is not going to be on the hot path (at least for now), I'll happily let someone else do the optimizations. Let's merge it in as-is.